spec/unit/s3/request_spec.rb

```diff
@@ -70,21 +45,21 @@
 
   context "path-style" do
     let(:bucket) { "mah-bucket" }
-    subject(:request) { described_class.new("SERVER_NAME" => "s3.amazonaws.com", "PATH_INFO" => "/#{bucket}#{key}") }
+    subject(:request) { described_class.new("SERVER_PORT" => "80", "SERVER_NAME" => "s3.amazonaws.com", "PATH_INFO" => "/#{bucket}#{key}") }
```

Comment:
`Rack::Request` assumes `env['SERVER_PORT']` always returns something. Otherwise its [#host](https://github.com/rack/rack/blob/ab172af1b63f0d8e91ce579dd2907c43b96cf82a/lib/rack/request.rb#L106-L109) method doesn't work properly.